### PR TITLE
DiscIO: Don't use PlainFileReader in VolumeDirectory

### PR DESCRIPTION
**Pull request by JosJuice**
Blobs are not for regular file reading.

### CODE REVIEW

**Comment by lioncash** [comment]
LGTM (waiting for buildbot to come back online)

**Comment by lioncash** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DiscIO/VolumeDirectory.cpp [hunk 1]
```diff
@@ -138,7 +137,7 @@ bool CVolumeDirectory::Read(u64 _Offset, u64 _Length, u8* _pBuffer, bool decrypt
 			if (_Length < fileBytes)
 				fileBytes = _Length;
 
-			if (!reader->Read(fileOffset, fileBytes, _pBuffer))
+			if (!file.Seek(fileOffset, SEEK_SET) || !file.ReadBytes(_pBuffer, fileBytes))
```
Comment (phire) [inline Source/Core/DiscIO/VolumeDirectory.cpp]: Could we have less obfuscated control flow here? 
I find things that take advantage of short-circuited evaluation are *needlessly* harder to understand.

    if (!file.Seek(...))
        return false;
    if (!file.ReadBytes(...))
        return false;

**Comment by phire** [comment]
Otherwise, LGTM

**Pull request merged**